src/components/anti-everyone.ts

```diff
@@ -31,9 +45,58 @@ export default class AntiEveryone extends BotComponent {
         if (message.content.match(failed_everyone_re) != null) {
             // NOTE: .toLocaleString("en-US") formats this number with commas.
             const memberCount = this.wheatley.TCCPP.members.cache.size.toLocaleString("en-US");
-            await message.reply({
+
+            // Store the reply for later deletion, if necessary
+            if (!this.replies.has(message.author)) {
+                this.replies.set(message.author, []);
+            }
+            const reply = await message.reply({
                 content: `Did you really just try to ping ${memberCount} people?`,
             });
+
+            // Store the reply for later deletion, along with the message it was replying to
+            this.replies.get(message.author)!.push(
+                Object.assign(
+                    {
+                        replyTo: message.id,
+                    },
+                    reply,
+                ),
+            );
+        }
+    }
+
+    /**
+     * Deletes all Replies that were made to a particular user
+     * @note this should be used to auto-hide spam message replies in order to try and reduce the effect of spam
+     * @TODO: Actually bind this into the anti-spam system
+     */
+    async deleteReplies(user: Discord.User) {
+        if (!this.replies.has(user)) {
+            return;
+        }
+
+        const deletedAll = Promise.all(this.replies.get(user)!.map(reply => reply.delete()));
+        this.replies.remove(user);
+        return deletedAll;
+    }
+
+    /**
+     * Auto-delete replies to messages that were deleted
+     * @param message The message that was deleted
```

Comment:
The comments are super appreciated though in this case the code is self-documenting